src/azure-cli/azure/cli/command_modules/monitor/_autoscale_util.py

```diff
@@ -557,6 +557,87 @@
+    }
+
+    # find the default profile and ensure that if there are multiple, they are consistent
+    default_profile = default_profiles[0] if default_profiles else None
+    print(default_profile.capacity.default.to_serialized_data())
+    print(default_profile.capacity.minimum.to_serialized_data())
+    print(default_profile.capacity.maximum.to_serialized_data())
+    for p in default_profiles:
+        _validate_default_profile(default_profile, p)
+
+    for profile in recurring_profiles:
+        # portal creates extra default profiles with JSON names...
+        # trying to stay compatible with that
+        try:
+            # portal-created "default" or end time
+            json_name = json.loads(profile.name.to_serialized_data())
+            sched_name = json_name['for']
+            end_time = time(hour=profile.recurrence.schedule.hours[0].to_serialized_data(),
+                            minute=profile.recurrence.schedule.minutes[0].to_serialized_data())
+
+            if not default_profile:
+                # choose this as default if it is the first
+                default_profile = {
+                    "name": 'default',
+                    "capacity": profile.capacity,
+                    "rules": profile.rules
+                }
+            else:
+                # otherwise ensure it is consistent with the one chosen earlier
+                _validate_default_profile(default_profile, profile)
+
+            for day in profile.recurrence.schedule.days:
+                day = day.to_serialized_data()
+                if day not in profile_schedule:
+                    profile_schedule[day] = {}
+                if sched_name in profile_schedule[day]:
+                    profile_schedule[day][sched_name]['end'] = end_time
+                else:
+                    profile_schedule[day][sched_name] = {'end': end_time}
+        except ValueError:
+            # start time
+            sched_name = profile.name.to_serialized_data()
+            start_time = time(hour=profile.recurrence.schedule.hours[0].to_serialized_data(),
+                              minute=profile.recurrence.schedule.minutes[0].to_serialized_data())
+            for day in profile.recurrence.schedule.days:
+                day = day.to_serialized_data()
+                if day not in profile_schedule:
+                    profile_schedule[day] = {}
+                if sched_name in profile_schedule[day]:
+                    profile_schedule[day][sched_name]['start'] = start_time
+                    profile_schedule[day][sched_name]['capacity'] = profile.capacity
+                    profile_schedule[day][sched_name]['rules'] = profile.rules
+                else:
+                    profile_schedule[day][sched_name] = {
+                        'start': start_time,
+                        'capacity': profile.capacity,
+                        'rules': profile.rules
+                    }
+
+    return default_profile, profile_schedule
```

Comment:
The default_profile here can be a Dict or a AAZObject. They are different.

Reply:
default_profile's format is accepted for instance property update